azext_iot/tests/dps/test_iot_dps_int.py

```diff
@@ -5,779 +5,763 @@
-                self.check("attestation.type.x509.clientCertificates.secondary", None),
-                self.exists("reprovisionPolicy"),
-                self.check("allocationPolicy", AllocationType.hashed.value),
-                self.check("reprovisionPolicy.migrateDeviceData", False),
-                self.check("reprovisionPolicy.updateHubAssignment", False),
-                self.check("capabilities.iotEdge", False),
-            ],
-        ).get_output_in_json()["etag"]
-
-        self.cmd(
-            "iot dps registration list -g {} --dps-name {} --enrollment-id {}".format(
-                rg, dps, enrollment_id
-            ),
-            checks=[self.check("length(@)", 0)],
-        )
-
-        cert_name = self.create_random_name("certificate-for-test", length=48)
-        cert_etag = self.cmd(
-            "iot dps certificate create -g {} --dps-name {} --name {} --p {}".format(
-                rg, dps, cert_name, cert_path
-            ),
-            checks=[self.check("name", cert_name)],
-        ).get_output_in_json()["etag"]
-
-        self.cmd(
-            "iot dps enrollment-group update -g {} --dps-name {} --enrollment-id {}"
-            " --cn {} --etag {} --allocation-policy {} --webhook-url {} --api-version {}".format(
-                rg,
-                dps,
-                enrollment_id,
-                cert_name,
-                etag,
-                AllocationType.custom.value,
-                webhook_url,
-                api_version,
-            ),
-            checks=[
-                self.check("attestation.type", AttestationType.x509.value),
-                self.check("enrollmentGroupId", enrollment_id),
-                self.check("allocationPolicy", "custom"),
-                self.check("customAllocationDefinition.webhookUrl", webhook_url),
-                self.check("customAllocationDefinition.apiVersion", api_version),
-                self.check("attestation.x509.caReferences.primary", cert_name),
-                self.check("attestation.x509.caReferences.secondary", None),
-            ],
-        )
-
-        self.cmd(
-            "iot dps enrollment-group delete -g {} --dps-name {} --enrollment-id {}".format(
-                rg, dps, enrollment_id
+        for auth_phase in DATAPLANE_AUTH_TYPES:
+            enrollment_id = self.generate_enrollment_names(group=True)[0]
+            etag = self.cmd(
+                self.set_cmd_auth_type(
+                    "iot dps enrollment-group create --enrollment-id {} -g {} --dps-name {}"
+                    " --cp {} --scp {} --provisioning-status {} --allocation-policy {}"
+                    " --iot-hubs {} --edge-enabled".format(
+                        enrollment_id,
+                        self.entity_rg,
+                        self.entity_dps_name,
+                        CERT_PATH,
+                        CERT_PATH,
+                        EntityStatusType.enabled.value,
+                        "geoLatency",
```

Comment:
Can we use `AllocationType.geolatency.value`? Is there an issue with casing?

Reply:
Fixed the enum to be geoLatency and changed the code a bit to allow for geolatency or geoLatency to be accepted. This way our customers should not be affected for now.